saml-handler/src/main/java/org/apache/sling/auth/saml2/AuthenticationHandlerSAML2.java

```diff
@@ -0,0 +1,661 @@
+import org.opensaml.messaging.decoder.MessageDecodingException;
+import org.opensaml.messaging.encoder.MessageEncodingException;
+import org.opensaml.saml.common.messaging.context.SAMLBindingContext;
+import org.opensaml.saml.common.messaging.context.SAMLEndpointContext;
+import org.opensaml.saml.common.messaging.context.SAMLPeerEntityContext;
+import org.opensaml.saml.saml2.binding.decoding.impl.HTTPPostDecoder;
+import org.opensaml.saml.saml2.binding.encoding.impl.HTTPRedirectDeflateEncoder;
+import org.opensaml.saml.saml2.core.*;
+import org.opensaml.saml.common.xml.SAMLConstants;
+import org.opensaml.saml.saml2.encryption.Decrypter;
+import org.opensaml.saml.saml2.metadata.Endpoint;
+import org.opensaml.saml.saml2.metadata.SingleSignOnService;
+import org.opensaml.saml.security.impl.SAMLSignatureProfileValidator;
+import org.opensaml.xmlsec.SignatureSigningParameters;
+import org.opensaml.xmlsec.context.SecurityParametersContext;
+import org.opensaml.xmlsec.encryption.support.DecryptionException;
+import org.opensaml.xmlsec.encryption.support.InlineEncryptedKeyResolver;
+import org.opensaml.xmlsec.keyinfo.impl.StaticKeyInfoCredentialResolver;
+import org.opensaml.xmlsec.signature.support.SignatureConstants;
+import org.opensaml.xmlsec.signature.support.SignatureException;
+import org.opensaml.xmlsec.signature.support.SignatureValidator;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.wiring.BundleWiring;
+import org.apache.sling.auth.core.spi.AuthenticationHandler;
+import org.apache.sling.auth.core.spi.AuthenticationInfo;
+import org.apache.sling.auth.core.spi.DefaultAuthenticationFeedbackHandler;
+import org.apache.sling.auth.saml2.sp.SessionStorage;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.jcr.RepositoryException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.math.BigInteger;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import static org.apache.sling.auth.saml2.idp.Saml2IDPServlet.TEST_IDP_ENDPOINT;
+
+
+@Component(
+        service = AuthenticationHandler.class ,
+        name = AuthenticationHandlerSAML2.SERVICE_NAME,
+        configurationPid = "org.apache.sling.auth.saml2.impl.SAML2ConfigServiceImpl",
+        configurationPolicy = ConfigurationPolicy.REQUIRE,
+        property = {"sling.servlet.methods={GET, POST}",
+            AuthenticationHandler.PATH_PROPERTY+"={}",
+            AuthenticationHandler.TYPE_PROPERTY + "=" + AuthenticationHandlerSAML2.AUTH_TYPE,
+            "service.description=SAML2 Authentication Handler",
+            "service.ranking=42",
```

Comment:
Any particular reason for this service.ranking value :-) ?

Reply:
haha sort of, yeah. When I first started, I could not get Sling to use this authHandler. I tried high and low numbers, but only the form handler would get used. it seems only path length really has an impact in terms of which authentication handler 'wins' Since the service ranking seemed ineffective, I set it to a low but unique number between 0 and 500 lol

Reply:
`service.ranking` is supposed to be an int! If it is a String is it not effective!

Reply:
Awesome! thanks for the feedback. That makes a big difference, so I think service.ranking should be in the configurations.